apps/recnet/src/server/routers/article.ts

```diff
@@ -12,13 +13,35 @@ export const articleRouter = router({
     .input(getArticlesParamsSchema)
     .output(getArticlesResponseSchema)
     .query(async (opts) => {
-      const { link } = opts.input;
+      const { link, useDigitalLibrary } = opts.input;
       const { recnetApi } = opts.ctx;
       const { data } = await recnetApi.get(`/articles`, {
         params: {
           link,
+          useDigitalLibrary,
         },
       });
       return getArticlesResponseSchema.parse(data);
     }),
+  // getDbArticleByLink: checkIsAdminProcedure
+  //   .input(getArticlesParamsSchema)
+  //   .output(getArticlesResponseSchema)
+  //   .query(async (opts) => {
+  //     const { link } = opts.input;
+  //     const { recnetApi } = opts.ctx;
+  //     const { data } = await recnetApi.get(`/articles/db`, {
+  //       params: {
+  //         link,
+  //       },
+  //     });
+  //     return getArticlesResponseSchema.parse(data);
+  //   }),
+  updateArticleByLink: checkIsAdminProcedure
+    .input(AdminUpdateArticleDtoSchema)
+    .mutation(async (opts) => {
+      const { link, ...data } = opts.input;
+      const { recnetApi } = opts.ctx;
+
+      await recnetApi.patch(`/articles/admin?link=${link}`, data);
+    }),
```

Comment:
Your current code doesn't allow the admin to modify the "link" field. To allow, you should change your code like this.
```suggestion
  updateArticleByLink: checkIsAdminProcedure
    .input(AdminUpdateArticleDtoSchema)
    .mutation(async (opts) => {
      const { link } = opts.input;
      const { recnetApi } = opts.ctx;

      await recnetApi.patch(`/articles/admin?link=${link}`, opts.input);
    }),
```